samcli/lib/cookiecutter/question.py

```diff
@@ -0,0 +1,126 @@
+""" This module represents the questions to ask to the user to fulfill the cookiecutter context. """
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+
+class QuestionKind(Enum):
+    """ An Enum of possible question types. """
+
+    info = "info"
+    confirm = "confirm"
+    choice = "choice"
+    default = "default"
+
+
+class Question:
+    """
+    A question to be prompt to the user in an interactive flow where the response is used to fulfill
+    the cookiecutter context.
+
+    Attributes
+    ----------
+    _key: str
+        The key of the cookiecutter config to associate the answer with.
+    _text: str
+        The text to prompt to the user
+    _options: Optional[List[str]]
+        if this is a mcq question, this attribute specifies the possible choices
+    _required: bool
+        Whether the user must provide an answer for this question or not.
+    _default_answer: Optional[str]
+        A default answer that is suggested to the user
+    _next_question_map: Optional[Dict[str, str]]
```

Comment:
and also what `*` means (default)

Reply:
* rethinking about the string casting, I think I will change the map to be `_next_question_map: Optional[Dict[Any, str]]` instead of casting
* the `*` means match any answer, I think this is a common behavior, right?

Reply:
> rethinking about the string casting, I think I will change the map to be _next_question_map: Optional[Dict[Any, str]] instead of casting

Reply:
> rethinking about the string casting, I think I will change the map to be _next_question_map: Optional[Dict[Any, str]] instead of casting

how it affects the load question from json? json key should be string right?

> the * means match any answer, I think this is a common behavior, right?

not until wildcard came to my mind... `*` could also be a possible user input. I would rather have a dedicate attribute `_default_next_question`

Reply:
* `Any` is the type of the answer not the key, _next_question_map = {answerX: qX, answerY: qY: *(every other answer): qZ}
* make sense

Reply:
discard my previous comment, answer has to be a string as well as it is also provided by the use in the questions.yaml file. I will add a comment why we do the casting.